### PR TITLE
feat: add drone sftp cache plugin.

### PR DESCRIPTION
**Pull request by appleboy**
cc @bradrydzewski

### CODE REVIEW

File: content/appleboy/drone-sftp-cache/index.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  rebuild_cache:
+    image: applebot/drone-sftp-cache
+    path: /var/cache/drone
+    rebuild: true
+    mount:
+      - node_modules
++   when:
++     status: success
++     event: tag
+```
```
Comment (bradrydzewski) [inline content/appleboy/drone-sftp-cache/index.md]: remove this example per [documentation guidelines](http://readme.drone.io/tutorials/how-to-register-a-plugin/#documentation-guidelines)

File: content/appleboy/drone-sftp-cache/index.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+```diff
+pipeline:
+  rebuild_cache:
+    image: applebot/drone-sftp-cache
+    path: /var/cache/drone
+    rebuild: true
+    mount:
+      - node_modules
++   when:
++     status: success
```
Comment (bradrydzewski) [inline content/appleboy/drone-sftp-cache/index.md]: remove this example per [documentation guidelines](remove this example per [documentation guidelines](http://readme.drone.io/tutorials/how-to-register-a-plugin/#documentation-guidelines)

File: content/appleboy/drone-sftp-cache/index.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+image: appleboy/drone-sftp-cache
+---
+
+Use this plugin for caching build artifacts to speed up your build times. This plugin can create and restore caches of any folders.
+
+```yaml
+pipeline:
+  restore_cache:
+    image: applebot/drone-sftp-cache
+    server: ${SFTP_CACHE_SERVER}
```
Comment (bradrydzewski) [inline content/appleboy/drone-sftp-cache/index.md]: Lets remove all `${...}`. Secrets should be documented separately in the official documentation. If secrets change (which they do in 0.6) we then have to go through and update all the plugin documentation.

Comment (appleboy) [inline content/appleboy/drone-sftp-cache/index.md]: done.

File: content/appleboy/drone-sftp-cache/index.md (hunk 2 above)
Comment (appleboy) [inline content/appleboy/drone-sftp-cache/index.md]: done.

File: content/appleboy/drone-sftp-cache/index.md (hunk 1 above)
Comment (appleboy) [inline content/appleboy/drone-sftp-cache/index.md]: done.

File: content/appleboy/drone-sftp-cache/index.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    username: test
+-   password: test
++   key: plain_text_of_user_private_key
+    path: /var/cache/drone
+    rebuild: true
+    mount:
+      - node_modules
+```
+
+Example configuration for ignoring creates a hash file name based on branch name:
```
Comment (tboerger) [inline content/appleboy/drone-sftp-cache/index.md]: What does that mean?

Comment (appleboy) [inline content/appleboy/drone-sftp-cache/index.md]: https://github.com/drone/drone-plugin-index/pull/40/commits/c08d5f443d5ea3446bae6bcefa3ea1eebe60c7e3

see the detail https://github.com/appleboy/drone-sftp-cache/blob/master/plugin.go#L99-L103

**Comment by appleboy** [comment]
@tboerger @bradrydzewski Any feedbacks?

**Pull request merged**